pip/index.py

```diff
@@ -828,7 +841,8 @@ def links(self):
                 url = self.clean_link(
                     urllib_parse.urljoin(self.base_url, href)
                 )
-                yield Link(url, self)
+                pyrequire = unescape(anchor.get('data-requires-python'))
```

Comment:
We need to check that the attribute is not `None` before passing it to `unescape`, because `unescape` doesn't like None.